integration/examples/jib-sync/README.md

```diff
@@ -0,0 +1,165 @@
+
+## Running the example
+
+Run the maven or gradle version of the example with port forwarding on.
+
+#### Gradle
+```
+$ skaffold dev -f skaffold-gradle.yaml --port-forward
+```
+
+#### Maven
+```
+$ skaffold dev -f skaffold-maven.yaml --port-forward
+```
+
+You can now see sync in action:
+1. See the original response from the `HelloController` in the spring application 
+  ```
+  $ curl localhost:8080
+  text-to-replace
+  ```
+1. Edit the hello controller `src/main/java/hello/HelloController.java`
+  ```diff
+  +       return "some-new-text";
+  -       return "text-to-replace";
+  ```
+1. Give skaffold a few seconds to synchronize the file to the container, and give Spring
+   Boot Developer Tools a chance to reload your application.
+1. See the updated response from the `HelloController`
+  ```
+  $ curl localhost:8080
+  some-new-text
+  ```
+1. You've now seen auto sync in action!
+
+## How it works
+
+This example contains both maven and gradle build configs and separate skaffold.yamls.
+
+*gradle*: `skaffold-gradle.yaml`
+*maven*: `skaffold-maven.yaml`
+
+use the `-f` flag to specify the correct buildfile when running (or rename your preferred option to `skaffold.yaml`)
+```
+$ skaffold -f skaffold-gradle.yaml ...
+```
+
+We configure it in `skaffold.yaml`, by enabling sync on the jib artifact.
+
+```yaml
+build:
+  artifacts:
+  - image: skaffold-example
+    context: .
+    jib: {}
+    sync: 
+      auto: {}
+```
+
+This example is designed around the functionality available in [Spring Boot Developer Tools](https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-devtools) for developing against running applications.
+
+Some additional steps in your java build are required for this to work:
+- Sync requires `tar` on the running container to copy files over. The default base image that Jib uses `gcr.io/distroless/java` does not include `tar` or any utilities. During development you must use a base image that includes `tar`, in this example we use the `debug` flavor of distroless: `gcr.io/distroless/java:debug` 
+`maven`
+```xml
+<plugin>
+  <groupId>com.google.cloud.tools</groupId>
+  <artifactId>jib-maven-plugin</artifactId>
+  <version>${jib.maven-plugin-version}</version>
+  <configuration>
+    ...
+    <from>
+      <image>gcr.io/distroless/java:debug</image>
+    </from>
+  </configuration>
+</plugin>
+```
+
+`gradle`
+```groovy
+jib {
+  ...
+  from {
+    image = "gcr.io/distroless/java:debug"
+  }
+}
+```
+
+- You must include the `spring-boot-devtools` dependency at the `compile/implementation` scope, which is contrary to the configuration outlined in the [official docs](https://docs.spring.io/spring-boot/docs/current/reference/html/using-spring-boot.html#using-boot-devtools). Because jib is unaware of any special spring only configuration in your builds, we recommend using profiles to turn on or off devtools support in your jib container builds.
```

Comment:
I think Jib currently includes `spring-boot-devtools` even if `<optional>true`. (And we decided to take care of this using an extension.) Should we still advise to set `<scope>compile`?

Reply:
I think we leave optional dependencies out. no?

Reply:
because it's not explicitly resolved to be a runtime dependency.

Reply:
oh about the scope, `<compile>` is default isn't it?

Reply:
Ah, right, `<compile>` is already the default. The Spring Boot doc only adds `<optional>true`, so this comment doesn't seem to apply.

And I think we include `spring-boot-devtools` regardless of `<optional>true`. (Would be good to double-check.)

Reply:
yeah you're right. I can say <!-- optional not necessary -->